### ISSUE TITLE
Chart won't display data

### ISSUE THREAD

**Issue by Cynnexis** [Cynnexis/Pinguin#2]
`PingChart` is an object that inherits from [`QChart`](https://doc.qt.io/qt-5.9/qchart.html). It has the method `append` to automatically add new points in a [`QLineSeries`](https://doc.qt.io/qt-5.11/qlineseries.html). However, when a new point is added, it does not appear in the graph. And as **all the points** are dynamically added, the graph shows nothing but the axes.

![bug2-chart](https://user-images.githubusercontent.com/2932967/46920946-b104e980-cfcb-11e8-93fd-d3b8b63a96a6.png)

However, the points are in the [`QLineSeries`](https://doc.qt.io/qt-5.11/qlineseries.html):

![bug2-points](https://user-images.githubusercontent.com/2932967/46920952-bfeb9c00-cfcb-11e8-99d0-3b02b805138f.png)

Maybe the problem comes from the fact that I don't really use a [`QChartView`](https://doc.qt.io/qt-5/qchartview.html) but one its parent, [`QGraphicsView`](http://doc.qt.io/qt-5/qgraphicsview.html), because I use QDesigner to create the form, and it does not have the [`QChartView`](https://doc.qt.io/qt-5/qchartview.html) widget yet, so I used [this solution from Stack Overflow](https://stackoverflow.com/a/48363007/7347145).

**Comment by Cynnexis** [comment]
When the [`QLineSeries`](https://doc.qt.io/qt-5.11/qlineseries.html) received the points **before** being set as default series to the chart, the data are shown:

```cpp
QLineSeries* ls_ping = new QLineSeries(this);

*ls_ping << QPointF(0, 20);
*ls_ping << QPointF(1, 40);
*ls_ping << QPointF(2, 30);

chart->addSeries(ls_ping);
chart->createDefaultAxes();
```

And the result is:

![bug2-chart](https://user-images.githubusercontent.com/2932967/46921210-d5ae9080-cfce-11e8-96d3-5e96990459f9.png)

One can notice the strange line on the right part of the chart. It goes from the final point `(2, 30)` to something like `(1.9, 40)`. The problem related to this *fourth* line is not the goal of this issue. It will be treated later if it's still exist once this problem is resolved.

When the *static points* are added *after* the link between the chart and the [`QLineSeries`](https://doc.qt.io/qt-5.11/qlineseries.html), the data is missing in the chart.

```cpp
QLineSeries* ls_ping = new QLineSeries(this);

chart->addSeries(ls_ping);
chart->createDefaultAxes();

*ls_ping << QPointF(0, 20);
*ls_ping << QPointF(1, 40);
*ls_ping << QPointF(2, 30);
```

**Comment by Cynnexis** [comment]
Fixed #2 by resizing the axis each time a point is added (with the help of signal/slot). I thought it would be automatic, but apparently not. `QtCharts` is hated by a lot of developers in the Qt community, and now I understand why... It looks like a beta version of what it already should be. This is very disappointing. Anyway, the following is the snippet of the solution:

```cpp
const int xeps = 1, yeps = 10; // epsilon
int xmin = 0;
int xmax = 10 + xeps;
int ymin = 0;
int ymax = 500 + yeps;

QValueAxis* axisX = new QValueAxis();
QValueAxis* axisY = new QValueAxis();

axisX->setRange(xmin, xmax);
axisY->setRange(ymin, ymax);

// [...]

void PingChart::onPointAdded(int index) {
	int x, y;
	x = round(ls_ping->points()[index].x());
	y = round(ls_ping->points()[index].y());

	if (x > xmax - xeps)
		xmax = x + xeps;

	if (y > ymax - yeps)
		ymax = y + yeps;

	axisX->setRange(xmin, xmax);
	axisY->setRange(ymin, ymax);
}
```

And the result is:

![bug2-finished](https://user-images.githubusercontent.com/2932967/46921791-06de8f00-cfd6-11e8-8258-423eb4edd872.png)

However, the Y-axis does not seems to fit back because the first value is too big. I should put a special condition for the first point (*calibrating method*).
